internal/kubernetes/kubernetes.go

```diff
@@ -32,16 +24,12 @@ type Kubernetes struct {
 	Config *rest.Config
 }
 
-func NewKubernetes(authToken, addr, ca string) (*Kubernetes, error) {
-	if authToken != "" && addr != "" && ca != "" {
-		k := &Kubernetes{Config: &rest.Config{}}
-		k.Config.Host = addr
-		k.Config.TLSClientConfig.CAData = []byte(ca)
-		k.Config.BearerToken = authToken
-		return k, nil
-	}
+func NewKubernetes() (*Kubernetes, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.WarnIfAllMissing = false
 
-	config, err := rest.InClusterConfig()
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+	config, err := kubeConfig.ClientConfig()
```

Comment:
I thought this only read the `kubectl` config file. Don't we still need `rest.InClusterConfig` for running this in a pod?

Reply:
I will double check and manual test ahead of any merge, but should be okay: https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/tools/clientcmd/merged_client_builder.go#L120

Reply:
Awesome! That is definitely easier